### DISCUSSION TITLE
How to change kconfig parameters for ESP32 (IRAM)

### DISCUSSION THREAD

**Discussion by danieldecesaro** [Sloeber/arduino-eclipse-plugin#1644] [Q&A] [answered]
Guys, I have a problem, I need to put a timer routine in IRAM and to do this I need to change the following parameters in sdkconfig:
CONFIG_GPIO_CTRL_FUNC_IN_IRAM=y
CONFIG_GPTIMER_ISR_HANDLER_IN_IRAM=y
CONFIG_GPTIMER_ISR_IRAM_SAFE=y
ONFIG_GPTIMER_CTRL_FUNC_IN_IRAM=y

How can I do this?

And how can I check that the routines I need are being loaded into IRAM?

**Comment by jantje** [comment]
I think these are ESP32 questions and not Sloeber questions.
Looks like this should contain all the info you need though 
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html

**Reply by danieldecesaro** [reply]
Hello
     I know which parameters I should modify, but I don't know where, in Sloeber, I should modify it!!!
     Can you guide me in which file I should make these changes? Which file will generate the sdkconfig that will be attached to the project? And how can I review this file in my project?

**Comment by danieldecesaro** [comment, marked as answer]
Problem solved, I used direct calls to gpio.c functions as advised by @igrr in this issue:
https://github.com/espressif/esp-idf/issues/2408

**Reply by jantje** [reply]
Great
